sklearn/decomposition/tests/test_pca.py

```diff
@@ -48,6 +52,61 @@ def test_pca(svd_solver, n_components):
     assert_allclose(np.dot(cov, precision), np.eye(X.shape[1]), atol=1e-12)
 
 
+def linear_operator_from_matrix(A):
+    return LinearOperator(
+        shape=A.shape,
+        dtype=A.dtype,
+        matvec=lambda x: A @ x,
+        rmatvec=lambda x: A.T @ x,
+        matmat=lambda X: A @ X,
+        rmatmat=lambda X: A.T @ X,
+    )
+
+
+def test_linear_operator_matmul():
+    A = np.array([[1, 2, 3], [4, 5, 6]])
+    B = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])
+    B = linear_operator_from_matrix(B)
+    expected_error = "Input operand 1 does not have enough dimensions"
+    with pytest.raises(ValueError, match=expected_error):
+        A @ B
+
+
+def test_linear_operator_reversed_matmul():
+    A = np.array([[1, 2, 3], [4, 5, 6]])
+    B = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])
+    B = linear_operator_from_matrix(B)
+    result = (B.T @ A.T).T
+    assert np.allclose(result, [[38, 44, 50, 56], [83, 98, 113, 128]])
+
+
+@pytest.mark.parametrize("density", [0.01, 0.05, 0.10, 0.30])
+@pytest.mark.parametrize("n_components", [1, 2, 3, 10, SPARSE_MAX_COMPONENTS])
+@pytest.mark.parametrize("format", ["csr", "csc"])
+@pytest.mark.parametrize("svd_solver", ["arpack"])
+@pytest.mark.parametrize(
+    "rtol", [1e-07]
+)  # , 1e-06, 1e-05, 1e-04, 1e-03, 1e-02, 1e-01, 1e-00])
+def test_pca_sparse(
+    global_random_seed, rtol, svd_solver, format, n_components, density
+):
+    if svd_solver in ["lobpcg", "arpack"] and n_components == SPARSE_MAX_COMPONENTS:
+        pytest.skip("lobpcg and arpack don't support full solves")
```

Comment:
What do you mean by "full solves"? Do you mean
```suggestion
        pytest.skip("lobpcg and arpack only can be used when dimensionality is reduced")
```

Reply:
What about:

```suggestion
        pytest.skip("lobpcg and arpack only support truncated solves")
```

It could also be fair to remove the full solve cases from this test?

Reply:
> It could also be fair to remove the full solve cases from this test?

Either that or using `with pytest.raises` with an appropriate check on the error message to make sure that it is informative for the user instead of using `pytest.skip`.

Reply:
I'll remove it, I think testing for an informative error message is covered by `test_pca_validation` for now.